packages/tempus/src/Tempus_SolutionHistory_decl.hpp

```diff
@@ -292,19 +312,33 @@ class SolutionHistory
     Teuchos::RCP<Interpolator<Scalar> > unSetInterpolator();
   //@}
 
+  /// Print information on States in the SolutionHistory.
   void printHistory(std::string verb="low") const;
 
+  /** \brief Initialize SolutionHistory
+   *
+   *  This function will check if all member data is initialized
+   *  and is consistent.  This function does not make member data
+   *  consistent, but just checks it.  This ensures it is inexpensive.
+   */
+  void initialize() const;
+
+  /// Return if SolutionHistory is initialized.
+  bool isInitialized() { return isInitialized_; }
+
 
 protected:
 
   std::string                               name_;
-  Teuchos::RCP<Teuchos::ParameterList>      shPL_;
   Teuchos::RCP<std::vector<Teuchos::RCP<SolutionState<Scalar> > > > history_;
   Teuchos::RCP<Interpolator<Scalar> >       interpolator_;
   StorageType                               storageType_;
   int                                       storageLimit_;
 
   Teuchos::RCP<SolutionState<Scalar> > workingState_; ///< The state being worked on
+
+  mutable bool isInitialized_;     ///< Bool if SolutionHistory is initialized.
```

Comment:
Why do you need mutable? Shouldn't the state of this flag should only change in non-const functions that change the state of the object?

Reply:
So here was the thought.  Please let me know if you have a better way to do this.

The initialize() function is const, so that the object is not changed during initialize() (e.g., no expensive operations are done: memory allocation, etc.).  However we need to set isInitialized_ = true, if everything is initialized.  Thus I made isInitialized_ mutable.  Thoughts?